### ISSUE TITLE
Can not select a country when adding a New Address

### ISSUE THREAD

**Issue by johnathon-walnut** [frappe/erpnext#32427]
### Information about bug

When attempting to place an order, I have to supply a Shipping Address, and while all other fields seem to allow input, the Country field forces you to select from a list - when selecting a country (i.e. United Kingdom) or if you type the country name it instantly clears once you click away/click the option. 

Below is a gif of how I am attemping to input the country, although I can't imagine this as user error.
![firefox_cycrEqdH8n](https://user-images.githubusercontent.com/36936588/193243751-e65aadba-d3cc-4abf-bf84-7860341e2db0.gif)

I am further convinced that this is a frappe issue due to there being an error in the browser's JavaScript console when attempting to pick a country.

### Module

ecommerce

### Version

Frappe version - 14
ERPNext Version - 14

### Installation method

manual install

### Relevant log output / Stack trace / Full Error Message.

```shell
Uncaught (in promise) TypeError: e is undefined
    yl number_format.js:205
    set_link_title link.js:97
    set_formatted_input link.js:84
    set_input data.js:243
    set_model_value base_control.js:282
    u base_control.js:233
    promise callback*frappe.run_serially/< dom.js:269
    run_serially dom.js:267
    u base_control.js:231
    validate_and_set_in_model base_control.js:251
    promise callback*validate_and_set_in_model base_control.js:251
    parse_validate_and_set_in_model link.js:123
    setup_awesomeplete link.js:385
    jQuery 2
    fire awesomplete.js:504
    select awesomplete.js:282
    click awesomplete.js:117
    bind awesomplete.js:477
    bind awesomplete.js:476
    i awesomplete.js:126
    setup_awesomeplete link.js:187
    make_input link.js:57
    t base_input.js:62
    refresh_input base_input.js:89
    refresh base_control.js:140
    attach_doc_and_docfields layout.js:437
    refresh layout.js:314
    make field_group.js:23
    make dialog.js:48
    <anonymous> dialog.js:17
    <anonymous> cart:1373
    jQuery 7
number_format.js:205:1
```

**Comment by dj12djdjs** [comment]
Yeah, looks like its passing `frappe.boot.link_title_doctypes` which is undefined at the time of calling `in_list`

![image](https://user-images.githubusercontent.com/29856401/193377914-188b26ec-c33a-4abd-aa59-39fabbfa49fd.png)

**Comment by johnathon-walnut** [comment]
I did look into this a little bit more, it seems to be an issue with frappe itself. A solution that allowed me to place an order was to just put 
```
if (!frappe.boot) {
	frappe.boot = {
		link_title_doctypes: ["ToDo"],
	};
} else {
	frappe.boot.link_title_doctypes = ["ToDo"];
}
```

into the browser console, which allowed me to setup an address. 

I found this when searching for where link_title_doctypes is set, and saw it in `cypress/integration/control_link.js`

Obviously this isn't ideal, but since I'm just testing this works for me at the moment. 

Not entirely sure what a proper fix for this would be.
